extensions/wikia/PageHeader/PageTitle.class.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace PageHeader;
+
+use \RequestContext;
+
+class PageTitle {
+
+	/* @var string */
+	public $title;
+
+	/* @var string */
+	public $prefix;
+
+	/* @var \WikiaGlobalRegistry */
+	private $wg;
+
+	/* @var int */
+	private $namespace;
+
+	const PREFIX_LESS_NAMESPACES = [ NS_MEDIAWIKI, NS_TEMPLATE, NS_CATEGORY, NS_FILE ];
+
+	/**
+	 * @param \WikiaApp $app
+	 */
+	public function __construct( $app ) {
+		$this->wg = $app->wg;
+		$this->MWTitle = RequestContext::getMain()->getTitle();
+
+		$this->namespace = $this->MWTitle->getNamespace();
+		$this->title = $this->handleTitle($app);
+		$this->prefix = $this->handlePrefix();
+	}
+
+	private function handleTitle( \WikiaApp $app ): string {
+		if (\WikiaPageType::isMainPage()) {
+			return $this->titleMainPage();
+		} else if (in_array($this->namespace, \BodyController::getUserPagesNamespaces())) {
+			return $this->titleUserPages();
+		} else if ($this->MWTitle->isTalkPage() || $this->shouldNotDisplayNamespacePrefix($this->namespace)) {
+			return htmlspecialchars($this->wg->Title->getText());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)